### PR TITLE
feat: implements request form

### PR DESCRIPTION
**Pull request by yetti**
Sends requests to FOLIO.

Still need to implement the collection times messaging based on the date.

### CODE REVIEW

File: app/helpers/request_helper.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  end
+
+  def supplements(holding)
+    holding["holdingsStatementsForSupplements"].pluck("note")
+  end
+
+  def indexes(holding)
+    holding["holdingsStatementsForIndexes"].pluck("note")
+  end
```
Comment (scott-yeadon) [inline app/helpers/request_helper.rb:30]: supps and indexes can also have statements as well as note - so might need the same logic as per items_issues_held

File: app/services/catalogue_services_client.rb [hunk 2]
```diff
@@ -25,6 +27,46 @@ def get_holdings(instance_id:)
...
+    conn = Faraday.new(url: ENV["CATALOGUE_SERVICES_API_BASE_URL"]) do |f|
+      f.request :authorization, "Bearer", @bearer_token
+      f.response :json
+    end
+
+    request_body = JSON.generate(requesterId: requester,
+      instanceId: request[:instance_id],
+      holdingsId: request[:holdings_id],
+      itemId: request[:item_id],
+      year: [request[:year]],
```
Comment (scott-yeadon) [inline app/services/catalogue_services_client.rb:53]: Note that the JSON property is 'yearCaption'

File: app/services/catalogue_services_client.rb [hunk 3]
```diff
@@ -25,6 +27,46 @@ def get_holdings(instance_id:)
...
+      enumeration: request[:enumeration],
+      chronology: request[:chronology],
+      barcode: request[:barcode],
+      notes: request[:notes])
```
Comment (scott-yeadon) [inline app/services/catalogue_services_client.rb:57]: Note that JSON property is 'patronComments'

File: app/services/catalogue_services_client.rb [hunk 4]
```diff
@@ -25,6 +27,46 @@ def get_holdings(instance_id:)
...
+    end
+
+    request_body = JSON.generate(requesterId: requester,
+      instanceId: request[:instance_id],
+      holdingsId: request[:holdings_id],
+      itemId: request[:item_id],
+      year: [request[:year]],
+      enumeration: request[:enumeration],
+      chronology: request[:chronology],
+      barcode: request[:barcode],
```
Comment (scott-yeadon) [inline app/services/catalogue_services_client.rb:56]: Note that barcode will be ignored for checkouts, it's generated by catalogue-services for serials and taken from Folio item metadata for monographs.

File: app/helpers/request_helper.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    most_recent = holding["holdingsStatements"].last
+
+    if most_recent.present?
+      if most_recent["statement"].present?
+        most_recent["statement"]
+      elsif most_recent["note"].present?
+        most_recent["enumeration"]
+      end
+    end
+  end
```
Comment (scott-yeadon) [inline app/helpers/request_helper.rb:12]: It is also possible for both a statement and a note to be present, as a note applying to the statement.

File: app/services/catalogue_services_client.rb (hunk 2 above)
Comment (yetti) [reply app/services/catalogue_services_client.rb]: I guess GitHub Copilot can't read what's open in another IDE even though I'm signed in. 🤷‍♀️  Fixed.

File: app/services/catalogue_services_client.rb (hunk 3 above)
Comment (yetti) [reply app/services/catalogue_services_client.rb]: Also a GitHub job...Fixed.

File: app/helpers/request_helper.rb (hunk 1 above)
Comment (yetti) [reply app/helpers/request_helper.rb:44]: I've written a helper method to just merge the statements and notes together into a single array for these fields.

File: app/services/catalogue_services_client.rb (hunk 4 above)
Comment (yetti) [reply app/services/catalogue_services_client.rb]: Removed the `barcode`.

File: app/helpers/request_helper.rb (hunk 5 above)
Comment (yetti) [reply app/helpers/request_helper.rb:10]: Also munged statements and notes together using the helper method.

**Review by scott-yeadon — APPROVED** [review]

**Pull request merged**